### PR TITLE
Add VPAL Engine driver.

### PR DESCRIPTION
**Pull request by idegtiarov**
Implement notifications on the Bridge Admin page for cases Engine
failures

### CODE REVIEW

File: bridge_adaptivity/module/engines/engine_vpal.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import logging
+import requests
+import urlparse
+
+from module.engines.interface import EngineInterface
+
+log = logging.getLogger(__name__)
+
+
+ACTIVITY_PARAMETERS = (
```
Comment (wowkalucky) [inline bridge_adaptivity/module/engines/engine_vpal.py]: Change to `ACTIVITY_PARAMS` for consistency?

File: bridge_adaptivity/module/engines/engine_vpal.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    def check_engine_response(engine_status, action=None, activity_name=None, status=200):
+        if engine_status == status:
+            log.debug("[VPAL Engine] activity {} is {} to the VPAL Engine.".format(activity_name, action))
+            return True
+        else:
+            log.error("[VPAL Engine] activity {} is not {} to the VPAL Engine.".format(activity_name, action))
+            return False
+
+    @staticmethod
+    def fulfil_payload(payload={}, instance_to_parse=None):
```
Comment (kunanit) [inline bridge_adaptivity/module/engines/engine_vpal.py]: Can we change "fulfil" to "fulfill"?

File: bridge_adaptivity/module/engines/engine_vpal.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        if isinstance(instance_to_parse, Activity):
+            params = ACTIVITY_PARAMETERS
+        elif isinstance(instance_to_parse, SequenceItem):
+            params = SEQUENCE_ITEM_PARAMS
+        else:
+            raise ValueError("Payload for the instance {} cannot be prepared".format(instance_to_parse))
+        for param in params:
+            if param == 'type':
+                atype = TYPES.get(getattr(instance_to_parse, 'atype'), 'generic')
+                payload[param] = atype
```
Comment (kunanit) [inline bridge_adaptivity/module/engines/engine_vpal.py]: does this give activity field "atype" to the engine as "type"?

File: bridge_adaptivity/module/engines/engine_vpal.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    To use VPAL engine with the Bridge for Adaptivity configure following parameters in the secure settings::
+
+        ENGINE_MODULE = 'module.engines.engine_vpal'
+
+        # ENGINE_DRIVER is a string with the name of driver class in the engine module
+        ENGINE_DRIVER = 'EngineVPAL'
+
+        # ENGINE_SETTINGS is a dict, with the initial params for driver initialization
+        ENGINE_SETTINGS = {
+            'BASE_URL': 'http://test.engine.vpal.io/engine/api',
```
Comment (kunanit) [inline bridge_adaptivity/module/engines/engine_vpal.py]: This should use the ENGINE_SETTINGS django settings value

File: bridge_adaptivity/module/engines/engine_vpal.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+    To use VPAL engine with the Bridge for Adaptivity configure following parameters in the secure settings::
+
+        ENGINE_MODULE = 'module.engines.engine_vpal'
+
+        # ENGINE_DRIVER is a string with the name of driver class in the engine module
+        ENGINE_DRIVER = 'EngineVPAL'
+
+        # ENGINE_SETTINGS is a dict, with the initial params for driver initialization
+        ENGINE_SETTINGS = {
```
Comment (kunanit) [inline bridge_adaptivity/module/engines/engine_vpal.py]: Does it make sense to add the authentication token here too? (see authentication notes on endpoint document)

File: bridge_adaptivity/module/engines/engine_vpal.py (hunk 3 above)
Comment (idegtiarov) [reply bridge_adaptivity/module/engines/engine_vpal.py]: Yes, it is.

File: bridge_adaptivity/module/engines/engine_vpal.py (hunk 2 above)
Comment (idegtiarov) [reply bridge_adaptivity/module/engines/engine_vpal.py]: Done

File: bridge_adaptivity/module/engines/engine_vpal.py (hunk 4 above)
Comment (idegtiarov) [reply bridge_adaptivity/module/engines/engine_vpal.py]: Actually, it is. Here is added example into doc strings, ENGINE is initialized in module.__init__.

Comment (kunanit) [reply bridge_adaptivity/module/engines/engine_vpal.py]: Oops didn't realize this was in the docstring - thanks

**Pull request merged**